File: src/main/java/aurocosh/divinefavor/common/custom_data/living/data/curse/CurseDataSerializer.java
package aurocosh.divinefavor.common.custom_data.living.data.curse;

import aurocosh.divinefavor.common.lib.interfaces.INbtSerializer;
import net.minecraft.nbt.NBTTagCompound;

public class CurseDataSerializer implements INbtSerializer<CurseData> {
    private static final String TAG_CURSE_COUNT = "CurseCount";

    @Override
    public void serialize(NBTTagCompound nbt, CurseData instance) {
        nbt.setInteger(TAG_CURSE_COUNT, instance.getCurseCount());
    }

    @Override
    public void deserialize(NBTTagCompound nbt, CurseData instance) {
        instance.setCurseCount(nbt.getInteger(TAG_CURSE_COUNT));
    }
}

File: src/main/kotlin/aurocosh/divinefavor/common/custom_data/living/data/curse/CurseDataSerializer.kt
package aurocosh.divinefavor.common.custom_data.living.data.curse

import aurocosh.divinefavor.common.lib.interfaces.INbtSerializer
import net.minecraft.nbt.NBTTagCompound

class CurseDataSerializer : INbtSerializer<CurseData> {

    override fun serialize(nbt: NBTTagCompound, instance: CurseData) {
        nbt.setInteger(TAG_CURSE_COUNT, instance.curseCount)
    }

    override fun deserialize(nbt: NBTTagCompound, instance: CurseData) {
        instance.curseCount = nbt.getInteger(TAG_CURSE_COUNT)
    }

    companion object {
        private const val TAG_CURSE_COUNT = "CurseCount"
    }
}
